src/forwardproblem.cpp

```diff
@@ -186,11 +190,58 @@ void ForwardProblem::handlePreequilibration()
         model->setFixedParameters(originalFixedParameters);
     }
     
+    updateAndReinitStatesAndSensitivities();
+}
+    
+void ForwardProblem::updateAndReinitStatesAndSensitivities()
+{
     model->fx0_fixedParameters(&x);
     solver->reInit(t, &x, &dx);
-    if (solver->getSensitivityOrder() >= SensitivityOrder::first &&
-        solver->getSensitivityMethod() == SensitivityMethod::forward)
-        solver->sensReInit( &sx, &sdx);
+    rdata->x0 = std::move(x.getVector());
+    if(solver->getSensitivityOrder() >= SensitivityOrder::first) {
+        model->fsx0_fixedParameters(&sx, &x);
+        
+        for (int ip = 0; ip < model->nplist(); ip++)
+            for (int ix = 0; ix < model->nx; ix++)
+                rdata->sx0[ip * model->nx + ix] = sx.at(ix, ip);
+        
+        if(solver->getSensitivityMethod() == SensitivityMethod::forward)
+            solver->sensReInit(&sx, &sdx);
+    }
+}
+    
+void ForwardProblem::handlePresimulation(int *ncheck)
```

Comment:
Do we need both preequilibration and presimulation? Wouldn't preequilibration just be the special case of presimulation with t_presim=Inf?

Reply:
We may want to to preequilibration AND presimulation at the same time. Long term this might be replaced by events that can also update fixed parameters, but nobody knows when that will be implemented ...